### ISSUE TITLE
Can't connect on Linux

### ISSUE THREAD

**Issue by bartinfinity** [spikeydragoon/Cross-Ark-Chat#59]
Hi what permissions are needed for this program to run properly? I start it on Linux but it doesn't work. Looks like it is not connecting to servers and I don't know why. I prepared config file on Windows and copied it - does configs are same for Linux and Windows?

![image](https://user-images.githubusercontent.com/2501014/124384444-8f934100-dcd1-11eb-9124-90fcdb6b85ee.png)

My current config:

```
//For all info on config visit https://github.com/spikeydragoon/Cross-Ark-Chat and look at the readme file as it goes over all config information.
// DO NOT change anything in the Rcon settings unless you know what you are doing!!!
{
  "ApplicationSettings": {
    "VersionCheckUrl": "https://raw.githubusercontent.com/spikeydragoon/Cross-Ark-Chat/master/version.txt",
    "PatchNotesCheckUrl": "https://raw.githubusercontent.com/spikeydragoon/Cross-Ark-Chat/master/CrossArkChatPatchNotes.txt",
    "LogErrors": true,
    "LogChat": true,
    "LogAdminCommands": true,
    "LogTribeLogs": true,
    "UseDiscord": false,
    "DiscordLogSeverity": "info"
  },

  "Servers": [
    {
      "NameTag": "[GlownySerwer]",
      "IP": "46.251.244.123",
      "RconPort": 15092,
      "QueryPort": 9778,
      "Password": "***********************",
      "SettingToUse": 1,
      "Active": true
    },
    {
      "NameTag": "[WypadowySerwer]",
      "IP": "46.251.241.170",
      "RconPort": 27092,
      "QueryPort": 21778,
      "Password": "***********************",
      "SettingToUse": 1,
      "Active": true
    }
  ],

  "ServerSettings": [
    {
      "SettingNumber": 1,
      "ServerTag": "",
      "ServerPlayerTag": ":",
      "PlayerChatTag": ":",
      "ChatTag": "",
      "HideNameTag": false,
      "DiscordChannelId": 0
    },
    {
      "SettingNumber": 2,
      "ServerTag": "",
      "ServerPlayerTag": ":",
      "PlayerChatTag": ":",
      "ChatTag": "",
      "HideNameTag": false,
      "DiscordChannelId": 0
    }
  ],

  "PrefixGroups": [
    {
      "ChatPrefix": "/prefix",
      "ServerList": [
        "chat1",
        "chat2"
      ]
    },
    {
      "ChatPrefix": "/prefix2",
      "ServerList": [
        "chat1",
        "chat2"
      ]
    }
  ],

  "Tribes": [
    {
      "TribeId": "00000",
      "SettingToUse": 1,
      "Active": false
    },
    {
      "TribeId": "00000",
      "SettingToUse": 2,
      "Active": false
    }
  ],

  "TribeSettings": [
    {
      "SettingNumber": 1,
      "ShowFullTribeLog": false,
      "ShowTribeName": false,
      "ShowServerName": true,
      "ShowTribeId": false,
      "EmbedTribeLog": true,
      "DiscordChannelId": 0
    },
    {
      "SettingNumber": 2,
      "ShowFullTribeLog": false,
      "ShowTribeName": false,
      "ShowServerName": true,
      "ShowTribeId": false,
      "EmbedTribeLog": true,
      "DiscordChannelId": 0
    }
  ],

  "GameCommands": [
    {
      "Command": "!discord",
      "CommandReply": "https://discord.gg/",
      "Active": false
    },
    {
      "Command": "rules",
      "CommandReply": "These are the rules.",
      "Active": false
    },
    {
      "Command": "/ping",
      "CommandReply": "Pong!",
      "Active": true
    }
  ],

  "CustomTagList": [
    "ACM[CMD]",
    "SERVER: ACM[CMD]"
  ],

  "RconSettings": {
    "NoResponceTag": "Server received, But no response!!",
    "NoPlayerTag": "No Players Connected",
    "ServerRepeatTag": "SERVER:",
    "AdminCMDTag": "AdminCmd:",
    "TribeLogTag": "Tribe",
    "GetChatCmd": "getchat",
    "ListPlayersCmd": "ListPlayers",
    "ListPlayerCheckTimeInMs": 10000,
    "GetChatTimeInMs": 1000,
    "PlayerCountCheckTimeInMs": 30000,
    "VersionCheckTimeInMs": 3600000,
    "RconSendTimeoutInMs": 3000,
    "RconReceiveTimeoutInMs": 10000,
    "RconConnectionRetrys": 1
  },

  "EmbeddedTribeLogColorSettings": [
    {
      "LogType": "killed",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    },
    {
      "LogType": "demolished",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    },
    {
      "LogType": "claimed",
      "LogColor": "03f413",
      "RoleToPing": "none"
    },
    {
      "LogType": "unclaimed",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    },
    {
      "LogType": "tamed",
      "LogColor": "03f413",
      "RoleToPing": "none"
    },
    {
      "LogType": "destroyed",
      "LogColor": "ff7400",
      "RoleToPing": "none"
    },
    {
      "LogType": "froze",
      "LogColor": "d8ddd8",
      "RoleToPing": "none"
    },
    {
      "LogType": "downloaded",
      "LogColor": "03f413",
      "RoleToPing": "none"
    },
    {
      "LogType": "uploaded",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    },
    {
      "LogType": "starved",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    }
  ],

  "ChatSettings": {
    "UseCustomTags": false,
    "SendChatToServers": true,
    "ShowAdminCommands": false,
    "UseGameCommands": false,
    "UsePrefixToSendChat": false,
    "UseWordReplacementList": false,
    "AllServersChatPrefix": "/all"
  },

  "DiscordSettings": {
    "DiscordBotSettings": {
      "CommandPrefix": "d!",
      "DiscordBotToken": "BotToken",
      "IgnoreOtherBots": true,
      "ShowDiscordNickName": true
    },

    "ServerChatSettings": {
      "ChatDiscordChannelID": 0,
      "SendServerChatToOwnChannel": false
    },

    "DiscordChatSettings": {
      "DiscordChatPrefix": "Discord",
      "SendChatToDiscord": true,
      "UsePrefixToSendToDiscord": false,
      "PrefixToSendToDiscord": "/discord"
    },

    "NotificationSettings": {
      "VersionDiscordChannelId": 0,
      "PatchNotesDiscordChannelId": 0,
      "JoinLeaveNotificationsDiscordChannelID": 0,
      "UseVersionChecker": false,
      "UsePatchNotesChecker": false,
      "UseJoinLeaveNotifications": false,
      "UsePlayerCountChannelUpdater": false,
      "SendJoinLeaveNotificationsToServerOwnChannel": false
    },

    "AdminCommandSettings": {
      "AdminCommandsDiscordChannelID": 0,
      "SendAdminCommandsToOwnChannel": false
    },

    "TribeLogSettings": {
      "TribeLogsDiscordChannelID": 0,
      "ShowTribelogs": false,
      "ShowFullTribeLog": false,
      "ShowTribeId": true,
      "ShowTribeName": true,
      "ShowServerName": true,
      "EmbedTribeLog": true,
      "SendTribeLogsToOwnChannel": false
    },

    "SupportSettings": {
      "SupportDiscordChannelID": 0,
      "UseSupportPrefix": false,
      "PingRoleName": false,
      "ReplyToSupportPing": true,
      "SupportPrefix": "/help",
      "SupportRoleToPing": "rolename",
      "SupportPingReply": "Your support ticket has been sent."
    }
  }
}

```

I tried everything, set 744 for all files in directory and nothing. I couldn't understand why on Windows I got folder logs while on linux not. I logged to account from which I'm trying to run it and it start but same as on root account.

**Comment by spikeydragoon** [comment]
@bartinfinity Yep config is same for linux as windows so they are interchangeable.
Main reason for it not connecting is normally rcon port/password entered is wrong or the server/antivirus is blocking the connection to the ark server.
I find best way to test this is to use the tnc command in powershell.

tnc ip -port rconPortHere
Example
tnc 192.168.1.8 -port 32333

Not sure what the linux eq of the powershell tnc is but im sure a quick google will get you something.
If tnc says tcptestsucceeded True using the rcon port then your all set nothing is blocking it. But if it says False then something on the ark server is blocking it and you will need to check with the hosting company to make sure they are not blocking connections on that port.

**Comment by bartinfinity** [comment]
That strange, it stopped working on windows too. It were working on weekend.
![image](https://user-images.githubusercontent.com/2501014/124590681-b5e4e800-de5b-11eb-85c1-c242905b5881.png)

Is something bad in my config?

```
//For all info on config visit https://github.com/spikeydragoon/Cross-Ark-Chat and look at the readme file as it goes over all config information.
// DO NOT change anything in the Rcon settings unless you know what you are doing!!!
{
  "ApplicationSettings": {
    "VersionCheckUrl": "https://raw.githubusercontent.com/spikeydragoon/Cross-Ark-Chat/master/version.txt",
    "PatchNotesCheckUrl": "https://raw.githubusercontent.com/spikeydragoon/Cross-Ark-Chat/master/CrossArkChatPatchNotes.txt",
    "LogErrors": true,
    "LogChat": true,
    "LogAdminCommands": true,
    "LogTribeLogs": true,
    "UseDiscord": false,
    "DiscordLogSeverity": "info"
  },

  "Servers": [
    {
      "NameTag": "[GlownySerwer]",
      "IP": "46.251.244.123",
      "RconPort": 15092,
      "QueryPort": 9778,
      "Password": "",
      "SettingToUse": 1,
      "Active": false
    },
    {
      "NameTag": "[WypadowySerwer]",
      "IP": "46.251.242.114",
      "RconPort": 19092,
      "QueryPort": 13778,
      "Password": "",
      "SettingToUse": 1,
      "Active": true
    }
  ],

  "ServerSettings": [
    {
      "SettingNumber": 1,
      "ServerTag": "",
      "ServerPlayerTag": ":",
      "PlayerChatTag": ":",
      "ChatTag": "",
      "HideNameTag": false,
      "DiscordChannelId": 0
    },
    {
      "SettingNumber": 2,
      "ServerTag": "",
      "ServerPlayerTag": ":",
      "PlayerChatTag": ":",
      "ChatTag": "",
      "HideNameTag": false,
      "DiscordChannelId": 0
    }
  ],

  "PrefixGroups": [
    {
      "ChatPrefix": "/prefix",
      "ServerList": [
        "chat1",
        "chat2"
      ]
    },
    {
      "ChatPrefix": "/prefix2",
      "ServerList": [
        "chat1",
        "chat2"
      ]
    }
  ],

  "Tribes": [
    {
      "TribeId": "00000",
      "SettingToUse": 1,
      "Active": false
    },
    {
      "TribeId": "00000",
      "SettingToUse": 2,
      "Active": false
    }
  ],

  "TribeSettings": [
    {
      "SettingNumber": 1,
      "ShowFullTribeLog": false,
      "ShowTribeName": false,
      "ShowServerName": true,
      "ShowTribeId": false,
      "EmbedTribeLog": true,
      "DiscordChannelId": 0
    },
    {
      "SettingNumber": 2,
      "ShowFullTribeLog": false,
      "ShowTribeName": false,
      "ShowServerName": true,
      "ShowTribeId": false,
      "EmbedTribeLog": true,
      "DiscordChannelId": 0
    }
  ],

  "GameCommands": [
    {
      "Command": "!discord",
      "CommandReply": "https://discord.gg/",
      "Active": false
    },
    {
      "Command": "rules",
      "CommandReply": "These are the rules.",
      "Active": false
    },
    {
      "Command": "/ping",
      "CommandReply": "Pong!",
      "Active": true
    }
  ],

  "CustomTagList": [
    "ACM[CMD]",
    "SERVER: ACM[CMD]"
  ],

  "RconSettings": {
    "NoResponceTag": "Server received, But no response!!",
    "NoPlayerTag": "No Players Connected",
    "ServerRepeatTag": "SERVER:",
    "AdminCMDTag": "AdminCmd:",
    "TribeLogTag": "Tribe",
    "GetChatCmd": "getchat",
    "ListPlayersCmd": "ListPlayers",
    "ListPlayerCheckTimeInMs": 10000,
    "GetChatTimeInMs": 1000,
    "PlayerCountCheckTimeInMs": 30000,
    "VersionCheckTimeInMs": 3600000,
    "RconSendTimeoutInMs": 3000,
    "RconReceiveTimeoutInMs": 10000,
    "RconConnectionRetrys": 1
  },

  "EmbeddedTribeLogColorSettings": [
    {
      "LogType": "killed",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    },
    {
      "LogType": "demolished",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    },
    {
      "LogType": "claimed",
      "LogColor": "03f413",
      "RoleToPing": "none"
    },
    {
      "LogType": "unclaimed",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    },
    {
      "LogType": "tamed",
      "LogColor": "03f413",
      "RoleToPing": "none"
    },
    {
      "LogType": "destroyed",
      "LogColor": "ff7400",
      "RoleToPing": "none"
    },
    {
      "LogType": "froze",
      "LogColor": "d8ddd8",
      "RoleToPing": "none"
    },
    {
      "LogType": "downloaded",
      "LogColor": "03f413",
      "RoleToPing": "none"
    },
    {
      "LogType": "uploaded",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    },
    {
      "LogType": "starved",
      "LogColor": "ff2121",
      "RoleToPing": "none"
    }
  ],

  "ChatSettings": {
    "UseCustomTags": false,
    "SendChatToServers": true,
    "ShowAdminCommands": false,
    "UseGameCommands": false,
    "UsePrefixToSendChat": false,
    "UseWordReplacementList": false,
    "AllServersChatPrefix": "/all"
  },

  "DiscordSettings": {
    "DiscordBotSettings": {
      "CommandPrefix": "d!",
      "DiscordBotToken": "BotToken",
      "IgnoreOtherBots": true,
      "ShowDiscordNickName": true
    },

    "ServerChatSettings": {
      "ChatDiscordChannelID": 0,
      "SendServerChatToOwnChannel": false
    },

    "DiscordChatSettings": {
      "DiscordChatPrefix": "Discord",
      "SendChatToDiscord": true,
      "UsePrefixToSendToDiscord": false,
      "PrefixToSendToDiscord": "/discord"
    },

    "NotificationSettings": {
      "VersionDiscordChannelId": 0,
      "PatchNotesDiscordChannelId": 0,
      "JoinLeaveNotificationsDiscordChannelID": 0,
      "UseVersionChecker": false,
      "UsePatchNotesChecker": false,
      "UseJoinLeaveNotifications": false,
      "UsePlayerCountChannelUpdater": false,
      "SendJoinLeaveNotificationsToServerOwnChannel": false
    },

    "AdminCommandSettings": {
      "AdminCommandsDiscordChannelID": 0,
      "SendAdminCommandsToOwnChannel": false
    },

    "TribeLogSettings": {
      "TribeLogsDiscordChannelID": 0,
      "ShowTribelogs": false,
      "ShowFullTribeLog": false,
      "ShowTribeId": true,
      "ShowTribeName": true,
      "ShowServerName": true,
      "EmbedTribeLog": true,
      "SendTribeLogsToOwnChannel": false
    },

    "SupportSettings": {
      "SupportDiscordChannelID": 0,
      "UseSupportPrefix": false,
      "PingRoleName": false,
      "ReplyToSupportPing": true,
      "SupportPrefix": "/help",
      "SupportRoleToPing": "rolename",
      "SupportPingReply": "Your support ticket has been sent."
    }
  }
}
```

**Comment by bartinfinity** [comment]
Disregard my latest message, it is working on Windows, I don't know why it didn't worked when I last time checked it, now I'm on server within the ark so maybe the server were somehow "sleeping" or Nitrado were performing some restarts.

However it is still not working on my dedicated linux server with Debian while I used exactly same config, on one screenshot I've windows instance thats works and linux that not.
![image](https://user-images.githubusercontent.com/2501014/124626533-0e79ac80-de7f-11eb-8953-55f44930b148.png)

It is dedicated linux server there are no antiviruses and nor provided is blocking anything, I've full control on this server and I don't have enabled firewall.

**Comment by bartinfinity** [comment]
Ok something really bad is happening on my server.
I started up debian under WSL, downloaded same pagckage and it started without issues and connected.
![image](https://user-images.githubusercontent.com/2501014/124631039-47b41b80-de83-11eb-87c3-144d1a209a3f.png)
I'm closing, it is not releated the program.

**Comment by bartinfinity** [comment]
I finally figure out WHY app could not connect to my servers.
I don't know if this is limitation of the app or ark servers but I figure out that app sometimes didn't wanted connect to my servers even on Windows - until I started the game and connected to my servers.
So the app somehow is sharing connection, I suppose ark servers won't let to connect any connections that are not playing at game.
Since my server is dedicated and located at hosting in different country servers won't let to connect.
I manage to somehow host this app on my synology NAS server which is running 24/7 after I started game and connected to servers. I hope app will not be disconnected when servers restart at night.
